MvvmCross/Droid/Droid/Platform/MvxAndroidSetupSingleton.cs

```diff
@@ -73,31 +71,29 @@ public virtual void InitializeFromSplashScreen(IMvxAndroidSplashScreenActivity s
             lock (LockObject)
             {
                 this._currentSplashScreen = splashScreen;
-
-                if (this._initializationStarted)
+                if (_initialized)
                 {
-                    if (this._initialized)
-                    {
-                        this._currentSplashScreen.InitializationComplete();
-                        return;
-                    }
-
+                    this._currentSplashScreen?.InitializationComplete();
                     return;
                 }
 
-                this._initializationStarted = true;
-            }
-
-            this._setup.InitializePrimary();
-            ThreadPool.QueueUserWorkItem(ignored =>
-            {
-                this._setup.InitializeSecondary();
-                lock (LockObject)
+                if (IsInitialisedTaskCompletionSource != null)
                 {
-                    this._initialized = true;
-                    this._currentSplashScreen?.InitializationComplete();
+                    return;
+                }
+                else
+                {
+                    IsInitialisedTaskCompletionSource = new TaskCompletionSource<bool>();
+                    this._setup.InitializePrimary();
+                    ThreadPool.QueueUserWorkItem(ignored =>
+                    {
+                        this._setup.InitializeSecondary();
+                        IsInitialisedTaskCompletionSource.SetResult(true);
```

Comment:
Important: Missing the lock around 91-93, to match 74-76. (See https://github.com/andyci/MvvmCross/blob/bdcb56b5c9261bbaf63205785c7b22f166aa32e2/MvvmCross/Droid/Droid/Platform/MvxAndroidSetupSingleton.cs#L91). This could result in two calls to _currentSplashScreen?.InitializationComplete().

Technically, I think this part should be called back onto the main thread instead of the background. Calling InitializationComplete (which essentially calls IMvxAppStart.Start() and launches a new activity, see https://github.com/MvvmCross/MvvmCross/blob/4.0/MvvmCross/Droid/Droid/Views/MvxSplashScreenActivity.cs#L82) from a background thread doesn't make sense to me.